### DISCUSSION TITLE
How to prevent maximized window to cover dockapps (i.e. reserve space on right corner of screen)

### DISCUSSION THREAD

**Discussion by h-g-s** [bbidulock/icewm#779] [Q&A]
Hi,

I'm happy that I can run dockapps inside IceWM.
However, when I maximize a window (see screenshot), the window covers these dockapps. 

How can I specify that maximized windows should not cover these dockapps on the right ?
Any configuration to reserve 64 pixels at the right of screen should help...

Thanks !
![scr](https://github.com/bbidulock/icewm/assets/7953594/83fb1e0a-96ec-4084-af5a-85f84744f699)

**Comment by gijsbers** [comment]
This should have been an issue and not a discussion.

Commit 3e634220 implements this enhancement when DockApps has "dock" and LimitByDockLayer=1.

Didn't know before that the support for DockApps in IceWM was appreciated.

Let me know if this works for you. Please use issues next time.

**Comment by h-g-s** [comment]
Thanks for the prompt reply and patch ! 

Tried the new option, but it seems that maximized windows are still covering the dock (screenshot below).     
One improvement I saw is that now the "high" option is respected.


![Screenshot_2024-07-06_17-36-42](https://github.com/bbidulock/icewm/assets/7953594/a6df796c-7731-4d0a-a29d-114c1b92ffed)

**Comment by gijsbers** [comment]
Your DockApps doesn't have "dock", as I mentioned earlier. That is required to get the DockApps container in the `Dock` layer and only then does the `LimitByDockLayer=1` take effect.

Also include commit ce6ba582.

**Comment by h-g-s** [comment]
Which dock do you recommend me to use ?    

Can I start only the window maker dock to be used in IceWM ?

**Comment by gijsbers** [comment]
No.

At the bottom of your image you have a `DockApps = "right high desktop"` line. Here `desktop` is the bottommost layer, but you need to put it in the `dock` layer. The icewm and icesh manpages explain layers.

Do `icewm --rewrite-preferences` to update the comment for `DockApps = ` preference. Also see the new manpage icewm-preferences.

**Comment by h-g-s** [comment]
Works perfectly! Thank you!
